### PR TITLE
caf: led_effect: Expand the blink effect with second off period param…

### PR DESCRIPTION
**Pull request by rakons**
This PR expands the blink effect with the possibility to add second
period time that would be used as an off time.
The way LED_COLOR works was updated to allow safely pass color argument
from one macro to another.

### CODE REVIEW

**Comment by rakons** [comment]
I added also some configuration in ML changes as a showcase. We can remove it from the PR as well if you wish to clean the PR.

**Comment by rakons** [comment]
We have 4 errors that relates to coding style, there are only 2 types of the errors I can see:
- Tabs after spaces - it was done to align with the rest of the code - We may agree we wish to fix it, but it would look odd if rest of the code stays as it is right now. So waiver or request to fix whole file here is requested.
- Requirement to put complex macros in brackets - we cannot do it! We need not to use the bracket here to be able to initialize structures as requested. We need a waiver here.

**Review by MarekPieta — APPROVED** [review]

File: applications/machine_learning/configuration/thingy52_nrf52832/led_state_def.h [hunk 1]
```diff
@@ -37,7 +37,7 @@ static const struct led_effect ei_data_forwarder_led_effects[] = {
 static const struct ml_result_led_effect ml_result_led_effects[] = {
```
[the remark this answers is not in the document]
Comment (rakons) [reply applications/machine_learning/configuration/thingy52_nrf52832/led_state_def.h:37]: Done

File: include/caf/led_effect.h [hunk 2]
```diff
@@ -95,7 +129,7 @@ struct led_effect {
 	{									\
 		.steps = ((const struct led_effect_step[]) {			\
 			{							\
-				.color = _color,				\
+				.color = LED_COLOR_USE(_color),			\
```
Comment (pdunaj) [inline include/caf/led_effect.h:132]: will effects defined by others have to use the new macro?

**Comment by rakons** [comment]
Changed just to add the new macro. Whole fixing of how LED_COLOR works removed as it would not be currently approved.

**Comment by rakons** [comment]
Just a simple grammar check.

**Review by pdunaj — APPROVED** [review]

**Comment by rakons** [comment]
Commit message line length fix.

**Comment by rakons** [comment]
Removing a leftover from previous idea - variadic color argument for LED_EFFECT_LED_BLINK2.

**Comment by rakons** [comment]
Rebase

**Comment by rakons** [comment]
Added changelog notes.

File: include/caf/led_effect.h (hunk 2 above)
Comment (rakons) [reply include/caf/led_effect.h]: Totally rebuilt right now.

File: include/caf/led_effect.h [hunk 3]
```diff
@@ -141,32 +162,50 @@ struct led_effect {
 		.loop_forever = false,						\
 	}
 
-/** Create LED blinking effect initializer.
+/** Create LED blinking effect initializer with two periods given
```
Comment (greg-fer) [inline include/caf/led_effect.h:165]:
```suggestion
/** Create LED blinking effect initializer with two periods as arguments.
```
Period missing.

File: include/caf/led_effect.h [hunk 4]
```diff
@@ -67,9 +67,30 @@ struct led_effect {
  */
 #define COLOR_BRIGHTNESS_TO_PCT(_val) ((_val * 100) / UINT8_MAX)
 
+/**
+ * @brief Pass a color value as a macro argument
+ *
+ * @ref LED_COLOR macro contains commas in its body.
+ * It means that after the argument is expanded it would be treated
```
Comment (greg-fer) [inline include/caf/led_effect.h:74]:
```suggestion
 * This means that after the argument is expanded, the macro is to be treated
```

File: include/caf/led_effect.h [hunk 5]
```diff
@@ -67,9 +67,30 @@ struct led_effect {
...
+ * as an multiple arguments. It makes it impossible to pass this argument
```
Comment (greg-fer) [inline include/caf/led_effect.h:75]:
```suggestion
* as multiple arguments. This also makes it impossible to pass an argument
```

File: include/caf/led_effect.h [hunk 6]
```diff
@@ -67,9 +67,30 @@ struct led_effect {
...
+ * from macro level to another macro.
+ * This macro allows the usage of the color argument as an argument to
```
Comment (greg-fer) [inline include/caf/led_effect.h:77]:
```suggestion
 * The macro allows the usage of the color argument as an argument to
```

File: include/caf/led_effect.h [hunk 7]
```diff
@@ -67,9 +67,30 @@ struct led_effect {
...
+ * another macro level
```
Comment (greg-fer) [inline include/caf/led_effect.h:78]: Period missing.

File: include/caf/led_effect.h [hunk 8]
```diff
@@ -67,9 +67,30 @@ struct led_effect {
...
+
 /** Create LED color initializer for LED turned on.
  *
  * @note As arguments, pass the brightness levels for every color channel.
+ * @note The macro returns structure initializer that after expanded
+ *       contains commas not placed in brackets.
+ *       It means that when passed as an argument this argument
+ *       cannot be passed simply to another macro.
+ *       Use @ref LED_COLOR_ARG_PASS macro for the preprocessor
+ *       to treat it as a single argument again.
```
Comment (greg-fer) [inline include/caf/led_effect.h:93]:
```suggestion
 * @note As arguments, pass the brightness levels for every color channel.
 * 
 * @note The macro returns the structure initializer that once expanded
 *       contains commas not placed in brackets.
 *       This means that when passed as an argument, this argument
 *       cannot be passed simply to another macro.
 *       Use @ref LED_COLOR_ARG_PASS macro for the preprocessor
 *       to treat it as a single argument again.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 9]
```diff
@@ -139,6 +139,11 @@ Common Application Framework (CAF)
 ----------------------------------
 
 * Migrated :ref:`nRF Desktop settings loader <nrf_desktop_settings_loader>` to :ref:`lib_caf` as :ref:`CAF: Settings loader module <caf_settings_loader>`.
+* Added:
+
+  * :ref"`led_effect_CAF`: New LED effect macro implemented: :c:macro:`LED_EFFECT_LED_BLINK2`.
+  * :ref"`led_effect_CAF`: Macro to pass color arguments between macro calls: :c:macro:`LED_COLOR_ARG_PASS`.
+
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:146]:
```suggestion
* :ref:`caf_leds`:

  * Added new LED effect macro: :c:macro:`LED_EFFECT_LED_BLINK2`.
  * Added a macro to pass color arguments between macro calls: :c:macro:`LED_COLOR_ARG_PASS`.

File: include/caf/led_effect.h (hunk 7 above)
Comment (rakons) [reply include/caf/led_effect.h:78]: Fixed

**Review by greg-fer — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Pull request merged**